src/frequenz/sdk/timeseries/battery_pool/_result_types.py

```diff
@@ -0,0 +1,131 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Methods for processing battery-inverter data."""
+
+from dataclasses import dataclass
+from datetime import datetime
+
+
+@dataclass
+class Bound:
+    """Lower and upper bound values."""
+
+    lower: float
+    """Lower bound."""
+
+    upper: float
+    """Upper bound."""
+
+
+@dataclass
+class CapacityMetrics:
+    """Capacity metrics."""
+
+    timestamp: datetime
+    """Timestamp of the metrics,"""
+
+    total_capacity: float
+    """Total batteries capacity.
+
+    Calculated with the formula:
+    ```
+    working_batteries: Set[BatteryData] # working batteries from the battery pool
+    total_capacity = sum(battery.capacity for battery in working_batteries)
+    ```
+    """
+    bound: Bound
+    """Capacity bounds.
+    
+    Bounds are calculated with the formula:
+    ```
+    working_batteries: Set[BatteryData] # working batteries from the battery 
+    bound.lower = sum(
+        battery.capacity * battery.soc_lower_bound for battery in working_batteries)
+
+    bound.upper = sum(
+        battery.capacity * battery.soc_upper_bound for battery in working_batteries)
+    ```
+    """
+
+
+@dataclass
+class SoCMetrics:
+    """Soc metrics."""
+
+    timestamp: datetime
+    """Timestamp of the metrics."""
+
+    average_soc: float
+    """Average soc.
+
+    Average soc is calculated with the formula:
+    ```
+    working_batteries: Set[BatteryData] # working batteries from the battery pool
+
+    used_capacity = sum(battery.capacity * battery.soc for battery in working_batteries)
+    total_capacity = sum(battery.capacity for battery in working_batteries)
+    average_soc = used_capacity/total_capacity
+    ```
+    """
+
+    bound: Bound
+    """SoC bounds weighted by capacity.
+    
+    Bounds are calculated with the formula:
+    capacity_lower_bound = sum(
+        battery.capacity * battery.soc_lower_bound for battery in working_batteries)
+    
+    capacity_upper_bound = sum(
+        battery.capacity * battery.soc_upper_bound for battery in working_batteries)
+
+    total_capacity = sum(battery.capacity for battery in working_batteries)
+    
+    bound.lower = capacity_lower_bound/total_capacity
+    bound.upper = capacity_upper_bound/total_capacity
+
+    """
+
+
+@dataclass
+class PowerBoundsMetrics:
```

Comment:
I wonder if it wouldn't be nicer for the user if this is called `PowerMetrics` instead, since then you have `bound` as memebers so it gets a bit redundant to say `PowerBoundsMetrics.supply_bound` for example. Also not sure if this could ever happen, but maybe in the future we need to add more power related metrics?

Reply:
Done